### PR TITLE
[RFR][DOC] Update dev guide from web_ui to NM+WT, initial progress

### PR DESCRIPTION
**Pull request by mfalesni**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 8c1d703bb21fa6f0f27a8e05be8caa926880fa58:

`docs/guides/ui_modeling.rst`:
- :grey_exclamation: Line 73: **INFO/1** *Duplicate implicit target name: "navmazing".*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: docs/guides/ui_modeling.rst [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+    * **NM**'s navigation returns the **WT** view of the last step.
+
+  * With the **WT** view returned from **NM** knowing that we indeed are on that particular
+    location, what usually happens next is that the ``create`` method dumps a dictionary of data
+    into view's fill method.
+
+    * **WT**'s view fill goes through the view and fills each widget that was specified in the
+      dictionary with the appropriate value. Widgets whose names weren't in the keys will be
+      skipped.
```
Comment (psav) [inline docs/guides/ui_modeling.rst]: None values are also skipped

**Comment by dajoRH** [comment]
Lint report for commit 5fc58945227ef2eb0fc009e368cd254e2c613602:

`docs/guides/ui_modeling.rst`:
- :grey_exclamation: Line 81: **INFO/1** *No role entry for "py:class" in module "docutils.parsers.rst.languages.en".*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c88e11e219f0b986e8532a263b5357fcbd0b7f37:

`docs/guides/ui_modeling.rst`:
- :grey_exclamation: Line 90: **INFO/1** *Duplicate implicit target name: "navmazing".*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by mfalesni** [comment]
^^ screw lint.

**Pull request merged**